onnx/test/large_onnx_test.py

```diff
@@ -0,0 +1,134 @@
+            large_model.save(filename)
+            copy = LargeModelContainer()
+            with self.assertRaises(RuntimeError):
+                assert copy.model_proto
+            copy.load(filename)
+            assert copy.model_proto is not None
+            checker.check_model(copy.model_proto)
+
+    @staticmethod
+    def _large_linear_regression():
+        X = make_tensor_value_info("X", TensorProto.FLOAT, [None, None])
+        Y = make_tensor_value_info("Y", TensorProto.FLOAT, [None])
+        graph = make_graph(
+            [
+                make_node("MatMul", ["X", "A"], ["XA"]),
+                make_node("MatMul", ["XA", "B"], ["XB"]),
+                make_node("MatMul", ["XB", "C"], ["Y"]),
+            ],
+            "mm",
+            [X],
+            [Y],
+            [
+                make_large_tensor_proto("#loc0", "A", TensorProto.FLOAT, (3, 3)),
+                from_array(np.arange(9).astype(np.float32).reshape((-1, 3)), name="B"),
+                make_large_tensor_proto("#loc1", "C", TensorProto.FLOAT, (3, 3)),
+            ],
+        )
+        onnx_model = make_model(graph)
+        large_model = make_large_model(
+            onnx_model.graph,
+            {
+                "#loc0": (np.arange(9) * 100).astype(np.float32).reshape((-1, 3)),
+                "#loc1": (np.arange(9) + 10).astype(np.float32).reshape((-1, 3)),
+            },
+        )
+        large_model.check_model()
+        return large_model
+
+    def test_large_one_weight_file(self):
+        large_model = self._large_linear_regression()
+        assert isinstance(large_model, LargeModelContainer)
+        with tempfile.TemporaryDirectory() as temp:
+            filename = os.path.join(temp, "model.onnx")
+            saved_proto = large_model.save(
+                filename, LargeModelFileFormat.SINGLE_TENSOR_FILE
+            )
+            assert isinstance(saved_proto, ModelProto)
+            copy = LargeModelContainer()
+            copy.load(filename)
+            copy.check_model()
+            copy2 = load_model(filename, load_external_data=True)
+            checker.check_model(copy2)
+
+    def test_large_onnx_multi_files(self):
+        large_model = self._large_linear_regression()
+        assert isinstance(large_model, LargeModelContainer)
+        with tempfile.TemporaryDirectory() as temp:
+            filename = os.path.join(temp, "model.onnx")
+            saved_proto = large_model.save(
+                filename, LargeModelFileFormat.ONE_TENSOR_PER_FILE
+            )
+            assert isinstance(saved_proto, ModelProto)
+            copy = load_model(filename)
+            checker.check_model(copy)
+            for tensor in _get_all_tensors(copy):
+                if uses_external_data(tensor):
+                    for ext in tensor.external_data:
+                        if ext.key == "location":  # type: ignore[attr-defined]
+                            assert os.path.exists(ext.value)
```

Comment:
Check that we ever hit the inside of the loop as well?

Reply:
done